### DISCUSSION TITLE
How to make an exe file read DLL files from a specific path?

### DISCUSSION THREAD

**Discussion by hoonlight** [pyinstaller/pyinstaller#7724] [Help] [answered]
I created a file using pyinstaller `--onefile` on windows.
In this case, the pytorch cuda related DLL files are very large, so I separated them instead of including them in the package.

And when I put them in the path where the exe file is located and run it, it works very well. It founds the DLL files.
But I want to create a "cuda" folder in the location of the exe file, put the DLL files in it, and make the exe recognize this path.

In short, I want the executable to look for dlls in the dist/cuda directory, not the dist directory. 
Including dlls in the executable is not a viable option in my case due to size.

```
if getattr(sys, "frozen", False):
    # running in a bundle
    current_dir = Path(sys.executable).parent
    cuda_dir = current_dir / "cuda"
    sys.path.append(str(cuda_dir))
```

I tried the above option, but it didn't work.
Is there any solution?
Please help

**Comment by rokm** [comment, marked as answer]
`sys.path` is search path for *python* modules.

To modify DLL search path, either modify the `PATH` environment variable, or use [`os.add_dll_directory`](https://docs.python.org/3/library/os.html#os.add_dll_directory).

**Reply by hoonlight** [reply]
[os.add_dll_directory](https://docs.python.org/3/library/os.html#os.add_dll_directory)

Works very well. Thank you very much.
